### PR TITLE
[PWA-2306]: Remove deprecated "id" field from customer queries

### PR DESCRIPTION
**Pull request by pedrochiossi**
## Description

PR to remove all `customer.id` fields in graphQL queries as it is deprecated. This also adds a merge function to Customer typePolicy so it will always merge incoming objects on Apollo cache, this change was necessary due to the lack of a unique identifier on `Customer` type. 

## Related Issue

Closes PWA-2306

## Acceptance

1. PWA Studio no longer requests the `id` field on `Customer` objects
2. `Customer` objects are still cached appropriately
3. There is no performance slowdown on authenticated routes like wishlist, saved payments, etc.

#### Test scenario(s) for direct fix/feature

1. Verify that creating an account still functions correctly (regression test)
2. Verify that signing in still functions correctly (regression test)
3. Verify that the Communications Page still functions correctly (regression test)
4. Verify that the Order History Page still functions correctly (regression test)
5. Verify that the Wishlist Page still functions correctly (regression test)
6. Verify that the Account Information Page still functions correctly (regression test)
7. Verify that the Checkout Page still functions correctly (regression test)
8. Verify that Address Book functions correctly
9. Verify that the Shipping Information component pulls accurate details from the Customer to pre-populate its fields
10. Verify that navigating around the site as an authenticated user functions correctly (regression test)

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/AccountInformationPage/accountInformationPage.gql.js [hunk 1]
```diff
@@ -20,18 +20,17 @@ export const CHANGE_CUSTOMER_PASSWORD = gql`
     ) {
         changeCustomerPassword(
             currentPassword: $currentPassword
-            newPassword: $newPassword
+            newPassword: $newPassword # eslint-disable-next-line @graphql-eslint/require-id-when-available
```
Comment (mikhaelbois) [inline packages/venia-ui/lib/components/AccountInformationPage/accountInformationPage.gql.js:23]: Only small thing is that the comment placement seems weird.
Could we have something like this?
```
changeCustomerPassword(
    currentPassword: $currentPassword
    newPassword: $newPassword
) # eslint-disable-next-line @graphql-eslint/require-id-when-available
{
    email
}
```

**Review by mikhaelbois — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/AccountInformationPage/accountInformationPage.gql.js (hunk 1 above)
Comment (pedrochiossi) [reply packages/venia-ui/lib/components/AccountInformationPage/accountInformationPage.gql.js:23]: I tried, but prettier does not like it.

**Review by mikhaelbois — APPROVED** [review]

**Comment by ericeoeur** [comment]
**Passed** 
- [ ] Verify that creating an account still functions correctly (regression test)
- [ ] Verify that signing in still functions correctly (regression test)
- [ ] Verify that the Communications Page still functions correctly (regression test)
- [ ] Verify that the Order History Page still functions correctly (regression test)
- [ ] Verify that the Wishlist Page still functions correctly (regression test)
- [ ] Verify that the Account Information Page still functions correctly (regression test)
- [ ] Verify that the Checkout Page still functions correctly (regression test)
- [ ] Verify that Address Book functions correctly
- [ ] Verify that the Shipping Information component pulls accurate details from the Customer to pre-populate its fields
- [ ] Verify that navigating around the site as an authenticated user functions correctly (regression test)

**Failed**
 ❌  Cypress test `e2eTests/checkout/registeredUserCheckout.spec.js` has failed after several attempts. 
image

@pedrochiossi Can you please review? Thank you!

**Comment by dpatil-magento** [comment]
@ericeoeur Can you pls try again, its fixed now.

image

**Comment by ericeoeur** [comment]
**Tests Passed** ✅
Create Account
Sign In
Address Book Page
Communications Page
Order History Page
Wishlist
Account Information Page
Checkout Page
Navigation
[Upgrade Test](https://jira.corp.magento.com/browse/PWA-1418)

@dpatil-magento and I have approved this PR and will be merging soon. Thanks!

**Pull request merged**